### PR TITLE
leaky relu speed

### LINKED ISSUE #10972: Gluon's PReLU is very slow and a fix to it
I have experienced significantly slow training speed when using PReLU activation instead of using ReLU activation with model composed using the gluon API. The speed of PReLU over ReLU is about 1/5 on GPU when measuring the number of samples processed per second.

Finally I managed to bring the speed of gluon's PReLU back to normal by the following modifications:
Following is the original init function of PReLU:

    def __init__(self, alpha_initializer=initializer.Constant(0.25), **kwargs):
        super(PReLU, self).__init__(**kwargs)
        with self.name_scope():
            self.alpha = self.params.get('alpha', shape=(1,), init=alpha_initializer)

Following is the modified init function of PReLU:

    def __init__(self, in_channels=1, alpha_initializer=initializer.Constant(0.25), **kwargs):
        super(PReLU, self).__init__(**kwargs)
        with self.name_scope():
            self.alpha = self.params.get('alpha', shape=(in_channels,), init=alpha_initializer)

The key is to pass in the expected number of channels to the PReLU block, so that it does not share the negative slope among the channels. The downside of this solution is that you need to pass in the number of channels every time.

I don't know why the two settings (shared vs. non-shared) have so drastically different performance. The contributors of mxnet should investigate this issue in further depth.

### PR DESCRIPTION
**Pull request by szha**
## Description ##
leaky relu speed. fixes #10972 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] use kernel launch for prelu forward
- [x] use kernel launch for prelu backward

### CODE REVIEW

**Comment by piiswrong** [comment]
Is there test coverage for this op?

**Comment by szha** [comment]
https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py#L618-L655

**Comment by piiswrong** [comment]
please fix lint

**Comment by szha** [comment]
will do once I finish the backward pass.

File: src/operator/leaky_relu-inl.h [hunk 1]
```diff
@@ -115,11 +116,27 @@ class LeakyReLUOp : public Operator {
       case leakyrelu::kPReLU: {
         weight = in_data[leakyrelu::kGamma].get<xpu, 1, DType>(s);
         if (weight.shape_.Size() == 1) {
-          Assign(out, req[leakyrelu::kOut],
-                 F<mshadow_op::xelu>(data, mshadow::expr::broadcast_scalar(weight, out.shape_)));
+          MXNET_ASSIGN_REQ_SWITCH(req[leakyrelu::kOut], Req, {
+            mxnet_op::Kernel<mxnet_op::op_with_req<op::mshadow_op::xelu, Req>, xpu>::Launch(
+              s, in_data[leakyrelu::kData].Size(), out_data[leakyrelu::kOut].dptr<DType>(),
+              in_data[leakyrelu::kData].dptr<DType>(), DType(weight[0]));
```
Comment (eric-haibin-lin) [inline src/operator/leaky_relu-inl.h]: Cannot access gpu ptr outside kernel

**Pull request merged**